plugins/nodes/src/js/data/__tests__/NodesNetworkClient-test.ts

```diff
@@ -0,0 +1,73 @@
+const mockRequest = jest.fn();
+jest.mock("@dcos/http-service", () => ({
+  request: mockRequest
+}));
+
+import { from } from "rxjs";
+import { marbles } from "rxjs-marbles/jest";
+import { take } from "rxjs/operators";
+
+import * as NodesNetworkClient from "../NodesNetworkClient";
+
+describe("Network Nodes Client", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("makes a request", () => {
+    mockRequest.mockReturnValueOnce(from([{}]));
+    NodesNetworkClient.fetchNodesNetwork();
+    expect(mockRequest).toHaveBeenCalled();
+  });
+
+  it(
+    "emits an event when the data is received",
+    marbles(m => {
+      const expectedResult = [
+        {
+          updated: "2019-02-04T14:17:12.697Z",
+          public_ips: ["18.185.33.115"],
+          private_ip: "10.0.6.192",
+          hostname: "ip-10-0-6-192"
+        }
+      ];
+      const expected$ = m.cold("--(j|)", {
+        j: {
+          response: expectedResult,
+          code: 200,
+          message: "OK"
+        }
+      });
+
+      mockRequest.mockReturnValueOnce(expected$);
+
+      const result$ = NodesNetworkClient.fetchNodesNetwork().pipe(take(1));
+      m.expect(result$).toBeObservable(expected$);
+    })
+  );
+
+  it(
+    "emits an error if non-2XX API response",
+    marbles(m => {
+      const mockResult$ = m.cold("--j", {
+        j: {
+          code: 500,
+          message: "Internal Server Error",
+          response: []
```

Comment:
Having empty array we actually test additional errors, it is just not super obvious.
What if we add a bogus error message there?

```
response: [
  "Backend error :("
]
```

Reply:
good catch! I added this response to the test case.